src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,604 @@
+    {
+        Rpp32s *srcPtrTemp = srcPtr;
+        Rpp32f *dstPtrTemp = dstPtr;
+
+        for(Rpp32u k = 0; k < srcGenericDescPtr->dims[level]; k++)
+        {
+            *dstPtrTemp++ = (((Rpp32f)(*srcPtrTemp + 128) - meanPtr[paramIdx]) * multiplierPtr[paramIdx]) + shift;
+            paramIdx += paramStride[level - 1];
+            *srcPtrTemp++;
+        }
+    }
+    else
+    {
+        for (int i = 0; i < *length; i++)
+        {
+            normalize_ND_tensor_nontoggle(srcPtr, srcGenericDescPtr, dstPtr, dstGenericDescPtr, meanPtr, multiplierPtr, shift, paramStride, length + 1, nDim - 1, level + 1, paramIdx);
+            paramIdx = (!paramStride[level - 1]) ? 0 : paramIdx + paramStride[level - 1];
+            dstPtr += dstGenericDescPtr->strides[level];
+            srcPtr += srcGenericDescPtr->strides[level];
+        }
+    }
+}
+
+void normalize_2D_tensor(Rpp32f *srcPtr, RpptGenericDescPtr srcDescPtr, Rpp32f *dstPtr, RpptGenericDescPtr dstDescPtr,
+                         Rpp32f *meanPtr, Rpp32f *invStdDevPtr, Rpp32f shift, Rpp32u *dims, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp = dstPtr;
+    Rpp32s paramIdx = 0;
+    for(Rpp32u i = 0; i < dims[0]; i++)
+    {
+        Rpp32f *srcPtrTempRow = srcPtrTemp;
+        Rpp32f *dstPtrTempRow = dstPtrTemp;
+        for(Rpp32u j = 0; j < dims[1]; j++)
+        {
+            *dstPtrTempRow++ = (*srcPtrTempRow++ - meanPtr[paramIdx]) * invStdDevPtr[paramIdx] + shift;
+            paramIdx += paramStride[0];
+        }
+        paramIdx = (!paramStride[1]) ? 0 : paramIdx + paramStride[1];
+        srcPtrTemp += srcDescPtr->strides[1];
+        dstPtrTemp += dstDescPtr->strides[1];
+    }
+}
+
+void normalize_2D_tensor_avx_axis2(Rpp32f *srcPtr, RpptGenericDescPtr srcDescPtr, Rpp32f *dstPtr, RpptGenericDescPtr dstDescPtr,
+                                   Rpp32f *meanPtr, Rpp32f *invStdDevPtr, Rpp32f shift, Rpp32u *dims, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp = dstPtr;
+    Rpp32s paramIdx = 0;
+    Rpp32u vectorIncrement = 8;
+    Rpp32u bufferLength = dims[1];
+    Rpp32u alignedLength = (bufferLength / 8) * 8;
+    Rpp32u numRows = dims[0];
+
+    __m256 pShift = _mm256_set1_ps(shift);
+    for(Rpp32u i = 0; i < numRows; i++)
+    {
+        Rpp32f *srcPtrTempRow = srcPtrTemp + i * srcDescPtr->strides[1];
+        Rpp32f *dstPtrTempRow = dstPtrTemp + i * dstDescPtr->strides[1];
+
+        // set mean and stddev
+        Rpp32f mean = meanPtr[i];
+        Rpp32f invStdDev = invStdDevPtr[i];
+        __m256 pMean, pInvStdDev;
+        pMean = _mm256_set1_ps(mean);
+        pInvStdDev = _mm256_set1_ps(invStdDev);
+
+        Rpp32u vectorLoopCount = 0;
+        for(; vectorLoopCount < alignedLength ; vectorLoopCount += 8)
+        {
+            __m256 pSrc = _mm256_loadu_ps(srcPtrTempRow);
+            __m256 pDst = _mm256_add_ps(_mm256_mul_ps(_mm256_sub_ps(pSrc, pMean), pInvStdDev), pShift);
```

Comment:
Did you check with murugan on this? We shoulc ideally consolidate this and use fmadd.

Reply:
Fixed this
missed this for 2d case initially.